### ISSUE TITLE
Keyboard interrupt in ert3 run

### ISSUE THREAD

**Issue by xjules** [equinor/ert#3099]
After some tests `KeyboardInterrupt` exception is not picked-up correctly in `_evaluator.py`, supposed way was like this:
```python
try:
    monitor.monitor(tracker)
except (KeyboardInterrupt):
       tracker.request_termination()
```

We should provide a correct exiting when the console run is interrupted.

**Comment by verveerpj** [comment]
I ran in a few issues that complicate the solution:
- In other places in the code, the signal handler for `SIGINT` is already set, without taking into account that this signal may also need to be handled elsewhere. This complicates signal handling since we need to make sure that all parts of the code that need to react to this signal can do this, preferably in a mutual independent manner.
- We use the `multiprocessing` in the Prefect code, and use `SIGINT` to kill the process spawned to stop an evaluation. As a result, this code will react when the user presses ctrl-c, and spits out exceptions to the console. This needs to be done in a more clean way.

These issues are not trivial to resolve. Therefore I created two issues to document them: #3134 and #3135. I  suggest that we postpone resolving this issue until these are resolved.
